src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -276,7 +276,11 @@ public override BoundNode Visit(BoundNode node)
 
                 if (!statement.WasCompilerGenerated)
                 {
-                    return CollectDynamicAnalysis(visited);
+                    // Do not instrument implicit constructor initializers
+                    if (!statement.IsConstructorInitializer() || statement.Syntax.Kind() != SyntaxKind.ConstructorDeclaration)
```

Comment:
We may find several such cases--code without source not marked WasCompilerGenetrated--and will need to handle them like this in an ad hoc fashion.